### PR TITLE
Add support for clientmetrics endpoint

### PR DESCRIPTION
**Pull request by Cruig**

### CODE REVIEW

File: packages/plugin-metrics/src/metrics.js [hunk 1]
```diff
@@ -6,11 +6,13 @@
 
 import {SparkPlugin} from '@ciscospark/spark-core';
 import Batcher from './batcher';
+import ClientMetricsBatcher from './clientMetricsBatcher';
```
Comment (ianwremmel) [inline packages/plugin-metrics/src/metrics.js]: filenames should be dasherized, not camelcase

File: packages/plugin-metrics/src/clientMetricsBatcher.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+ */
+
+import {
+  Batcher,
+  SparkHttpError
+} from '@ciscospark/spark-core';
+
+const sym = Symbol(`metric id`);
+
+const ClientMetricsBatcher = Batcher.extend({
```
Comment (ianwremmel) [inline packages/plugin-metrics/src/clientMetricsBatcher.js]: Can most of this code be eliminated by extending MetricsBatcher?

File: packages/plugin-metrics/test/unit/spec/clientMetricsBatcher.js [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    promise = promise.then(() => {
+      return promiseTick(1);
+    });
+    count -= 1;
+  }
+  return promise;
+}
+
+describe(`plugin-metrics`, () => {
+  describe(`ClientMetricsMetricsBatcher`, () => {
```
Comment (ianwremmel) [inline packages/plugin-metrics/test/unit/spec/clientMetricsBatcher.js]: too many metrics

File: packages/plugin-metrics/src/metrics.js (hunk 1 above)
Comment (Cruig) [inline packages/plugin-metrics/src/metrics.js]: fixed

File: packages/plugin-metrics/test/unit/spec/clientMetricsBatcher.js (hunk 3 above)
Comment (Cruig) [inline packages/plugin-metrics/test/unit/spec/clientMetricsBatcher.js]: fixed

File: packages/plugin-metrics/src/clientMetricsBatcher.js (hunk 2 above)
Comment (Cruig) [inline packages/plugin-metrics/src/clientMetricsBatcher.js]: fixed

**Comment by Cruig** [comment]
@ianwremmel Is this good to go now?

File: packages/plugin-metrics/src/client-metrics-batcher.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+const ClientMetricsBatcher = Batcher.extend({
+  namespace: `Metrics`,
+
+  prepareItem(item) {
+    // Add more defaults to payload when the clientmetrics endpoint evolves to support richer payloads
+    return Promise.resolve(item);
+  },
+
+  prepareRequest(queue) {
```
Comment (ianwremmel) [inline packages/plugin-metrics/src/client-metrics-batcher.js]: You can just resolve with queue here, you don't need to copy anything.

Comment (Cruig) [inline packages/plugin-metrics/src/client-metrics-batcher.js]: fixed

**Pull request merged**